packages/trpc/routers/admin.ts

```diff
@@ -25,13 +26,21 @@ export const adminAppRouter = router({
       const [
         [{ value: numUsers }],
         [{ value: numBookmarks }],
-        pendingCrawls,
+        [{ value: pendingCrawls }],
+        [{ value: failedCrawls }],
         pendingIndexing,
         pendingOpenai,
       ] = await Promise.all([
         ctx.db.select({ value: count() }).from(users),
         ctx.db.select({ value: count() }).from(bookmarks),
-        LinkCrawlerQueue.getWaitingCount(),
+        ctx.db
+          .select({ value: count() })
+          .from(bookmarkLinks)
+          .where(eq(bookmarkLinks.crawlStatus, "pending")),
```

Comment:
If we're going to do the 2d table, I think we can differentiate between the pending in database and pending in redis. I'm afraid of some jobs getting lost before getting enqueued to redis for example (or redis data getting wiped) which might cause confusion. So I think for both crawling and inference we can do:

```
crawlingStats: {
  waitingInRedis: number,
  pending: number,
  failed: number,
},
inference: {
 ... // Same
}
```

What do you think?

Reply:
I would leave this into an upcoming commits honestly. 
I have a few ideas regarding the job queue in general I might discuss with you tomorrow.
For now I think it's enough, what do you think ?

Reply:
Yeah, let's leave it for later then, I don't mind 👍